tests/integration/test_charm.py

```diff
@@ -120,3 +124,57 @@ async def test_get_primary_action(ops_test: OpsTest) -> None:
 
         # assert get-primary returned the right primary
         assert identified_primary == expected_primary
+
+
+async def test_exactly_one_primary_reported_by_juju(ops_test: OpsTest) -> None:
+    """Tests that there is exactly one replica set primary unit reported by juju"""
+    
+    async def get_unit_messages():
+        """Collects unit status messages"""
+        app = await app_name(ops_test)
+        unit_messages = {}
+
+        async with ops_test.fast_forward():
+            time.sleep(20)
+        
+        for unit in ops_test.model.applications[app].units:
+            unit_messages[unit.entity_id] = unit.workload_status_message
+
+        return(unit_messages)
```

Comment:
According to [the docs](https://github.com/charmed-kubernetes/pytest-operator/blob/main/docs/reference.md):
`ops_test.fast_forward():` "Temporarily speed up update-status firing rate for the current model. Returns an async context manager that temporarily sets update-status firing rate to fast_interval. If provided, when the context exits the update-status firing rate will be set to slow_interval. Otherwise, it will be set to the previous value."

Since it is used as a [context manager](https://book.pythontips.com/en/latest/context_managers.html) here it will only fast forward things that are indented underneath it. (What you have here works, but I think this might look a little nicer). Try this instead:
```suggestion
        async with ops_test.fast_forward(fast_interval=f"{STATUS_INTERVAL}s"):
            time.sleep(STATUS_INTERVAL)
            for unit in ops_test.model.applications[app].units:
                unit_messages[unit.entity_id] = unit.workload_status_message

        return(unit_messages)
```
with STATUS_INTERVAL as a constant at the top of the time STATUS_INTERVAL=5